python/ray/air/execution/_internal/actor_manager.py

```diff
@@ -0,0 +1,773 @@
+        kill: bool = False,
+    ) -> None:
+        """Remove a tracked actor.
+
+        If the actor has already been started, this will stop the actor. This will
+        trigger the :meth:`TrackedActor.on_stop
+        <ray.air.execution._internal.tracked_actor.TrackedActor.on_stop>`
+        callback once the actor stopped.
+
+        If the actor has only been requested, but not started, yet, this will cancel
+        the actor request. This will not trigger any callback.
+
+
+        If ``kill=True``, this will use ``ray.kill()`` to forcefully terminate the
+        actor. Otherwise, graceful actor deconstruction will be scheduled after
+        all currently tracked futures are resolved.
+
+        Args:
+            tracked_actor: Tracked actor to be removed.
+            kill: If set, will forcefully terminate the actor instead of gracefully
+                scheduling termination.
+        """
+        if tracked_actor in self._live_actors_to_ray_actors_resources:
+            # Ray actor is running.
+
+            if not kill:
+                # Schedule __ray_terminate__ future
+                ray_actor, _ = self._live_actors_to_ray_actors_resources[tracked_actor]
+
+                def on_actor_stop(*args, **kwargs):
+                    self._actor_stop_resolved(tracked_actor=tracked_actor)
+
+                stop_future = ray_actor.__ray_terminate__.remote()
+                self._actor_state_events.track_future(
+                    future=stop_future,
+                    on_result=on_actor_stop,
+                    on_error=on_actor_stop,
+                )
+                self._tracked_actors_to_state_futures[tracked_actor].add(stop_future)
+
+            else:
+                # kill = True
+                self._live_actors_to_kill.add(tracked_actor)
+
+        elif tracked_actor in self._pending_actors_to_attrs:
+            # Actor is pending, stop
+            _, _, resource_request = self._pending_actors_to_attrs.pop(tracked_actor)
+            self._resource_manager.cancel_resource_request(
+                resource_request=resource_request
+            )
+        else:
+            raise ValueError(f"Unknown tracked actor: {tracked_actor}")
+
+    def is_actor_started(self, tracked_actor: TrackedActor) -> bool:
+        """Returns True if the actor has been started.
+
+        Args:
+            tracked_actor: Tracked actor object.
+        """
+        return tracked_actor in self._live_actors_to_ray_actors_resources
+
+    def get_actor_resources(
+        self, tracked_actor: TrackedActor
+    ) -> Optional[AcquiredResources]:
+        """Returns the acquired resources of an actor that has been started.
+
+        This will return ``None`` if the actor has not been started, yet.
+
+        Args:
+            tracked_actor: Tracked actor object.
+        """
+        if not self.is_actor_started(tracked_actor):
+            return None
+
+        return self._live_actors_to_ray_actors_resources[tracked_actor][1]
```

Comment:
at a high level, we seem to maintain a bunch of local states.
I wonder if we can simplify these a bit, so we only maintain a list of actors, plus a dict of futures?
again, just a random thought.